stellar/stellar-cli
# Fix tests

### What
Several small fixes to tests:
- Change the version of soroban-token-sdk to match soroban-sdk in the test_token test-wasms.
- Use correct register contract function based on the version of sdk.
- Reference xdr types via the CLI re-export instead of sdk when sdk not imported.
- Move the help docs check into the main test run and simplify it.

### Why
The contract imports both the token sdk and the sdk at different versions. Because the token sdk depends on a specific version of the sdk two sdk's end up being imported into the contract. Two sdk's cannot be imported into the same contract because they both provide a panic_impl and rust only allows one to be linked.

This test failure was somehow accidentally introduced in:
- https://github.com/stellar/stellar-cli/pull/1646

In the other cases those other bugs have been merged while the tests were broken by the first thing. They weren't discovered till now because with the tests build broken the compiler was not pointing us at them.

The help test run was moved because it's so far it causes a complete job to occur before the test job gets a change to have its one be identified as a dependency. We haven't seen this same thing on other repos with multiple complete jobs, so not sure if this is new behaviour for GHA, or just something to do with timing on this repo. In either case, we have other generated checks in the main workflow and it doesn't hurt to put it there.

The help job got simplified because it was using github scripts, and github outputs, when we have a simpler pattern for doing that type of diff check already in this repo.
